tests/test_max_stable_set.py

```diff
@@ -304,13 +304,16 @@ def test_solving_problem_with_isolated_catchments(mocker, network, network_spati
                        network_spatial_tree=network_spatial_tree,
                        modes={'car', 'bus'})
     mss.solve()
-    assert mss.solution == {'stop_1': 'link_1_2_bus', 'stop_2': 'link_4_5_car', 'stop_3': 'link_7_8_car'}
+    non_deterministic_link_choices = ['link_1_2_bus', 'link_1_2_car', 'link_2_3_car']
+    assert mss.solution['stop_1'] in non_deterministic_link_choices
+    chosen_link = mss.solution['stop_1']
+    assert mss.solution == {'stop_1': chosen_link, 'stop_2': 'link_4_5_car', 'stop_3': 'link_7_8_car'}
```

Comment:
I would expect a purely linear model to always pick the same values, even if different choices lead to the same objective function value (i.e., a flat solution space). But this might not be the case when the variables are binary... Anyway, most likely a change in the way CBC works by default (you could try it with an older version of CBC?).

Could also update this example to have different coefficient values so we can be certain as to the optimal values.

Reply:
ok, I mocked it so there is a clear preference and only one acceptable solution 👍